lsr_fingerprint.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, Red Hat, Inc.
+# SPDX-License-Identifier: MIT
+"""
+Modify fingerprint in spec file to be system_role:$rolename
+
+lsr_fingerprint.py -
+scans files in the templates dir under ./role["lsrrolename"],
+if the file contains a string role["reponame"]:role["rolename"],
+replaces it with system_role:role["lsrrolename"].
+
+E.g., in metrics, "performancecopilot:ansible-pcp" is replaced with
+"system_role:metrics" in roles/bpftrace/templates/bpftrace.conf.j2.
+"""
+
+from os import listdir, walk
+from os.path import join, isfile, isdir
+
+roles = [
+    {
+        "reponame": "willshersystems",
+        "rolename": "ansible-sshd",
+        "lsrrolename": "sshd",
+    },
+    {
+        "reponame": "performancecopilot",
+        "rolename": "ansible-pcp",
+        "lsrrolename": "metrics",
+    },
+]
+
+# dirs in the current dir
+dirs = [d for d in listdir(".") if isdir(d)]
+for d in dirs:
+    for role in roles:
+        if role["lsrrolename"] == d:
+            for root, dirs, files in walk(d):
+                for dir in dirs:
```

Comment:
Is it fine that `dirs` repeats in the main parent loop and in this child loop?

Reply:
Hi @spetrosi!

The main loop (at line 35) is for selecting `metrics` and `sshd`, and the child loop is for `sshd/templates/` and
```
metrics/roles/bpftrace/templates/	metrics/roles/pcp/templates/
metrics/roles/elasticsearch/templates/	metrics/roles/redis/templates/
metrics/roles/grafana/templates/	metrics/roles/repository/templates/
metrics/roles/mssql/templates/		metrics/roles/spark/templates/
metrics/roles/pcp/meta/
```
in each role. As we have to support the vendored roles in `metrics`, the child loop is recursive.

Thanks!